tests/settings/ConfigurationGatheringTest.py

```diff
@@ -222,6 +224,53 @@ def test_targets(self):
 
         self.assertEqual(targets, ['cli', 'test1', 'test2'])
 
+    def test_targets_substring(self):
+        current_dir = os.path.abspath(os.path.dirname(__file__))
+        test_dir = os.path.join(current_dir, 'section_manager_test_files')
+        with change_directory(test_dir):
+            log_printer = ListLogPrinter()
+            _, _, _, targets = gather_configuration(
+                lambda *args: True,
+                log_printer,
+                arg_list=['-c', 'inherit_coafile', 'python'])
+            self.assertEqual(
+                targets, ['python', 'all.python', 'all.python.codestyle'])
+
+            log_messages = [_l.message for _l in log_printer.logs]
+            msg = "No matching section found for 'python'."
+            self.assertNotIn(msg, log_messages)
```

Comment:
This is not possible because there are already warning like `No bears matching 'LineLengthBear' were found. Make sure you have coala-bears installed or you have typed the name correctly.`